enhancements/insights/container-scanner.md

```diff
@@ -0,0 +1,498 @@
+
+## Design Details
+
+### Open Questions [optional]
+
+#### Container Scanner Image Availability
+
+The Insights Operator must be able to deploy the container scanner from an image that is available by default in the cluster image registry.
+
+How can we configure a _related image_ in the Insights Operator from its `ClusterVersion` What is the equivalent of a ClusterServiceVersion (CSV)’s `.spec.relatedImages` for OpenShift cluster operators?
+
+## Test Plan
+
+### Container Scanner Test Plan
+
+The container scanner can be tested in isolation to verify its capability to scan containers and collect the expected runtime info.
+
+Testing the container scanner can be done outside of the Insights Operator test suite.
+It requires integration tests with applications based on the detectable runtime stacks.
+
+### Insights Operator Test Plan
+
+The integration tests for the Insights Operator will verify that it is able to properly deploy the container scanner as a `DaemonSet` and have some smoke tests to validate that the additional runtime info can effectively be collected.
+
+## Graduation Criteria
+
+TBD
+
+### Dev Preview -> Tech Preview
+
+* Make the container scanner disabled by default
+* Add a non-public flag to enable it on a cluster to collect runtime information
+* End User documentation is not required
+* Sufficient test coverage
+* Gather feedback from users
+* Generate reports to validate the consumption of the additional runtime information
+
+### Tech Preview -> GA
+
+*  More testing
+*  Sufficient time for feedback
+*  Available by default
+*  Conduct load testing
+* User facing documentation created in [openshift-docs](https://github.com/openshift/openshift-docs/blob/main/support/remote_health_monitoring/using-insights-operator.adoc) if needed
+
+### Removing a deprecated feature
+
+Not relevant for this proposal
+
+## Upgrade / Downgrade Strategy
+
+Upgrades/downgrades of the container scanner will be handled by updating its image URL in the Insights Operator configuration.
+The release cycle of the container scanner is bound to the Insight Operator release cycle.
+
+This proposal does not bring additional change to the upgrade/downgrade of the OpenShift Insights Operator.
+
+## Version Skew Strategy
+
+This proposal does not impact the version skew strategy of the Insights Operator.
+
+## Operational Aspects of API Extensions
+
+The container scanner runs with its own cgroups and does not impact the scanned containers.
+
+However, it will consume a certain amount of CPU and memory on each worker node while it is scanning.
+These consumptions must be bounded to avoid disrupting the user workload.
+
+Scanning containers will also increase the time to collect insights by the operator. The workload gatherer is performed every 12 hours and its execution will depend on the number of containers that are scanned.
+	
+In the existing initial implementation of the container scanner, it takes __~180 ms to scan a single container__.
```

Comment:
It would be good to have some estimate (or average) on number of containers in node and also some estimate on CPU and memory consumption on node.